M/Matio/build_tarballs.jl

```diff
@@ -0,0 +1,62 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "Matio"
+version = v"1.5.23"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/tbeu/matio", "e9e063e08ef2a27fcc22b1e526258fea5a5de329")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+
+cd matio
+
+# trying with cmake
+
+
+cmake . \
+        -DBUILD_SHARED_LIBS:BOOL=ON \
+        -DMATIO_SHARED:BOOL=ON \
+	-DMATIO_DEFAULT_FILE_VERSION=7.3 \
+        -DMATIO_MAT73:BOOL=ON \
+        -DCMAKE_INSTALL_PREFIX=${prefix} \
+        -DMATIO_WITH_HDF5:BOOL=ON \
+        -DMATIO_WITH_ZLIB:BOOL=ON \
+        -DHDF5_ROOT:PATH=${prefix} \
+        -DHDF5_DIR:PATH=${prefix} \
+        -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN}
+
+cmake --build .
+cmake --install .
+
```

Comment:
Please build inside a subdirectory.  CMake makes it incredibly inconvenient to build in the same directory where the source code is
```suggestion
cd matio
mkdir build && cd build
cmake .. \
        -DBUILD_SHARED_LIBS:BOOL=ON \
        -DMATIO_SHARED:BOOL=ON \
	-DMATIO_DEFAULT_FILE_VERSION=7.3 \
        -DMATIO_MAT73:BOOL=ON \
        -DCMAKE_INSTALL_PREFIX=${prefix} \
        -DMATIO_WITH_HDF5:BOOL=ON \
        -DMATIO_WITH_ZLIB:BOOL=ON \
        -DHDF5_ROOT:PATH=${prefix} \
        -DHDF5_DIR:PATH=${prefix} \
        -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN}
make -j${nproc}
make install
```